### ISSUE TITLE
CS7035 should report the bad value

### ISSUE THREAD

**Issue by nkolev92** [dotnet/roslyn#67117]
**Version Used**:  17.5/17.6

**Steps to Reproduce**:

1. Create a project with the following content:
```cs
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <AssemblyVersion>1.0.0.0-bla</AssemblyVersion>
  </PropertyGroup>

</Project>

```

1. Run build. 

**Expected Behavior**:

> C:\Code\Temp\roslyn\obj\Debug\net7.0\roslyn.AssemblyInfo.cs(15,59): warning CS7035: The specified version string, `1.0.0.0-bla` does not conform to the recommended format - major.minor.build.revision [C:\Code\Temp\roslyn\roslyn.csproj] 

**Actual Behavior**:

> C:\Code\Temp\roslyn\obj\Debug\net7.0\roslyn.AssemblyInfo.cs(15,59): warning CS7035: The specified version string does not conform to the recommended format - major.minor.build.revision [C:\Code\Temp\roslyn\roslyn.csproj] 

Note that the example I'm using is not really how we hit the problem. 
We hit the problem with the autogenerated assemblyinfo and it's hard to see what the actual problem is there. 
Example: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=7399095&view=results

**Comment by jcouv** [comment]
Fix was merged into 17.6p2
